book/src/pong-tutorial/pong-tutorial-03.md

```diff
@@ -116,190 +116,328 @@ We're finally ready to implement the `PaddleSystem` in `systems/paddle.rs`:
 
-We create a unit struct `PaddleSystem`, and with the `SystemDesc` derive. This
-is short for **System Descriptor**. In Amethyst, systems may need to access
-resources from the `World` in order to be instantiated. For each `System`, an
-implementation of the `SystemDesc` trait must be provided to specify the logic
-to instantiate the `System`. For `System`s that do not require special
-instantiation logic, the `SystemDesc` derive automatically implements the
-`SystemDesc` trait on the system type itself.
-
-Next, we implement the `System` trait for it with the lifetime of the components
-on which it operates. Inside the implementation, we define the data the system
-operates on in the `SystemData` tuple: `WriteStorage`, `ReadStorage`, and
-`Read`. More specifically, the generic types we've used here tell us that the
-`PaddleSystem` mutates `Transform` components, `.write_component::<Transform>`, it
-reads `Paddle` components,.read\_component::<Paddle>()`, and also accesses the `InputHandler`resource we created earlier, using the`Read\`
-structure.
-
-> For `InputHandler`, make sure the parameter type is the same
-> as the one used to create the `InputBundle` earlier.
+We create a unit struct `PaddleSystem` and implement the `System` trait for it.
+The `System` trait is provided by legion and contains just one method, `build`:
+
+```rust
+# use amethyst::prelude::*;
+// legion's System trait
+pub trait System {
+    /// builds the Runnable part of System
+    fn build(self) -> Box<dyn ParallelRunnable + 'static>;
+}
+```
+
+Inside `build`, we  describe the data our System needs, and our System's behavior.
+Ultimately, `build` returns a `Runnable` struct, which legion uses during runtime 
+to run our System.
+
+> ### What's a `Box<dyn ParallelRunnable>`?
+> The `Runnable` trait exposes a `run` method, which legion invokes for us during runtime. 
+> A `ParallelRunnable` is just a `Runnable` which is also `Send` and `Sync`. Finally, to 
+> support any type which implements `ParallelRunnable`, the `build` method returns a `Box<dyn ParallelRunnable>`.
+
+First, we call `SystemBuilder::new` with a name: `"PaddleSystem"`. (Our system's name
+is just for debugging and visualization purposes.) 
+
+Second, we invoke `.with_query` with the argument `<(&Paddle, &mut Transform)>::query()`. In
+this context, the type `<(&Paddle, &mut Transform)>` declares a **view**. A view describes
+the kind of data that our system needs. This view says that our system reads `Paddle`
+components and mutates `Transform` components. Legion provides the `query`
+method to construct a query from the view that we've declared.
+
+Third, we invoke `.read_resource::<InputHandler>()`. This method declares that our system reads
+the `InputHandler` resource, so that we can read the input state.
+
+Finally, we invoke `build` with a closure. This closure accepts four arguments: `|_commands, world, input, query_paddles|`. We're not using `_commands` for now. The `world` contains our 
```

Comment:
Not sure if we want to cover the command buffer here. (I don't understand it myself yet)